src/pocketmine/plugin/PluginDescription.php

```diff
@@ -149,6 +161,50 @@ public function getCommands(){
 		return $this->commands;
 	}
 
+	/**
+	 * @return array
+	 */
+	public function getRequiredExtensions(){
+		return $this->extensions;
+	}
+
+	/**
+	 * Checks if the current PHP runtime has the extensions required by the plugin.
+	 *
+	 * @throws PluginException if there are required extensions missing or have incompatible version, or if the version constraint cannot be parsed
+	 */
+	public function checkRequiredExtensions(){
+		foreach($this->extensions as $name => $versionConstrs){
+			if(!extension_loaded($name)){
+				throw new PluginException("Required extension $name not loaded");
+			}
+
+			if(!is_array($versionConstrs)){
```

Comment:
from what I can understand from the code that loads this information, this condition will always be false, but because of the lack of proper type annotation, it's non-obvious.